scanners/nodejs/nodejs.go

```diff
@@ -0,0 +1,183 @@
+func (scanner *Scanner) DetectPlatform(searchDir string) (bool, error) {
+	scanner.packageJsonPath = checkPackageJSON(searchDir)
+	if scanner.packageJsonPath == "" {
+		log.TPrintf("Platform not detected")
+		return false, nil
+	}
+
+	scanner.nodeVersion = checkNodeVersion(searchDir)
+	scanner.packageManager = checkPackageManager(searchDir)
+	scanner.scripts = checkPackageScripts(scanner.packageJsonPath)
+
+	if slices.Contains(scanner.scripts, "build") {
+		log.TPrintf("- build - found")
+		scanner.buildScript = "build"
+	} else {
+		log.TPrintf("- build - not found")
+	}
+
+	if slices.Contains(scanner.scripts, "lint") {
+		log.TPrintf("- lint - found")
+		scanner.lintScript = "lint"
+	} else {
+		log.TPrintf("- lint - not found")
+	}
+
+	if slices.Contains(scanner.scripts, "test") {
+		log.TPrintf("- test - found")
+		scanner.testScript = "test"
+	} else {
+		log.TPrintf("- test - not found")
+	}
+
+	log.TSuccessf("Platform detected")
+
+	return true, nil
+}
+
+func (scanner *Scanner) ExcludedScannerNames() []string {
+	return []string{}
+}
+
+// Options returns the options for the scanner
+func (scanner *Scanner) Options() (models.OptionNode, models.Warnings, models.Icons, error) {
+	projectRootOption := models.NewOption(projectRootDirInputTitle, projectRootDirInputSummary, projectRootDirInputEnvKey, models.TypeSelector)
+	nodeVersionOption := models.NewOption(nodeVersionInputTitle, nodeVersionInputSummary, nodeVersionInputEnvKey, models.TypeOptionalUserInput)
+	packageManagerOption := models.NewOption(packageManagerInputTitle, packageManagerInputSummary, packageManagerInputEnvKey, models.TypeSelector)
+
+	if scanner.packageManager != "" {
+		configOption := models.NewConfigOption("node-js-"+scanner.packageManager+"-config", nil)
+		packageManagerOption.AddConfig(scanner.packageManager, configOption)
+	} else {
+		for _, pkgMgr := range pkgManagers {
+			configOption := models.NewConfigOption("node-js-"+pkgMgr.name+"-config", nil)
+			packageManagerOption.AddConfig(pkgMgr.name, configOption)
+		}
+	}
+
+	projectRootOption.AddOption(models.UserInputOptionDefaultValue, nodeVersionOption)
+	nodeVersionOption.AddOption(models.UserInputOptionDefaultValue, packageManagerOption)
+
+	return *projectRootOption, models.Warnings{}, nil, nil
+}
+
+// DefaultOptions returns the default options for the scanner
+func (scanner *Scanner) DefaultOptions() models.OptionNode {
+	projectRootOption := models.NewOption(projectRootDirInputTitle, projectRootDirInputSummary, projectRootDirInputEnvKey, models.TypeUserInput)
+	nodeVersionOption := models.NewOption(nodeVersionInputTitle, nodeVersionInputSummary, nodeVersionInputEnvKey, models.TypeOptionalUserInput)
+	packageManagerOption := models.NewOption(packageManagerInputTitle, packageManagerInputSummary, packageManagerInputEnvKey, models.TypeSelector)
+
+	for _, pkgMgr := range pkgManagers {
+		configOption := models.NewConfigOption("default-node-js-"+pkgMgr.name+"-config", nil)
```

Comment:
Can we reuse the `configName` utility function you created for the identifier creation here?